geode-core/src/test/java/org/apache/geode/cache/query/internal/types/TypeUtilsJUnitTest.java

```diff
@@ -0,0 +1,1197 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.cache.query.internal.types;
+
+import static org.apache.geode.cache.query.internal.parse.OQLLexerTokenTypes.*;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.Serializable;
+import java.lang.reflect.Field;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.PriorityQueue;
+import java.util.Random;
+import java.util.Vector;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import org.apache.geode.InternalGemFireError;
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.query.QueryService;
+import org.apache.geode.cache.query.TypeMismatchException;
+import org.apache.geode.cache.query.internal.CompiledValue;
+import org.apache.geode.cache.query.internal.parse.OQLLexerTokenTypes;
+import org.apache.geode.internal.cache.AbstractRegion;
+import org.apache.geode.internal.cache.BucketRegion;
+import org.apache.geode.internal.cache.LocalRegion;
+import org.apache.geode.internal.cache.PartitionedRegion;
+import org.apache.geode.pdx.internal.PdxInstanceEnum;
+import org.apache.geode.pdx.internal.PdxString;
+import org.apache.geode.test.junit.categories.UnitTest;
+
+@Category(UnitTest.class)
+@PowerMockIgnore("*.UnitTest")
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(TypeUtils.class)
+public class TypeUtilsJUnitTest {
```

Comment:
We're currently naming unit tests FooTest and integration tests FooIntegrationTest. No need for "JUnit" in the name of the class.

Reply:
Hey @kirklund,

In a previous pull request I was specifically asked by a reviewer to change the name of one of my tests from "FooTest" to "FooJUnitTest", that's why I create the tests using that suffix now. Moreover, within the [Code Contributions](https://cwiki.apache.org/confluence/display/GEODE/Code+contributions), specifically in the [Writing tests](https://cwiki.apache.org/confluence/display/GEODE/Writing+tests) section, we have the following:

```
JUnit tests should

. Use Junit 4 Syntax
. End with JUnitTest
. Contain an Category annotation of either UnitTest or IntegrationTest. UnitTests as should complete in milliseconds and test a specific class.
```

Which approach/guideline is the correct one?, are all reviewers on the same page regarding this?.
In the end, should I change the class name from `TypeUtilsJUnitTest` to `TypeUtilsTest` for this pull request to be accepted?. If the answer is "yes", shouldn't we change the Wiki and make sure all reviewers are in sync?.

Reply:
Hi Juan, I've brought this up on the dev-list.

Reply:
Hey @kirklund ,
Great!. I'll wait for the final decision and change the test name if needed.
Cheers.